lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -111,6 +111,36 @@ export class AuthorizationCodeModule extends AuthModule {
     }
 
     async acquireToken(request: TokenExchangeParameters): Promise<TokenResponse> {
+        const acquireTokenAuthority = (request && request.authority) ? AuthorityFactory.createInstance(request.authority, this.networkClient) : this.defaultAuthorityInstance;
+
+        let tokenEndpoint: string;
+        try {
+            tokenEndpoint = acquireTokenAuthority.tokenEndpoint;
+        } catch (e) {
+            const authErr: AuthError = e;
+            if (authErr.errorCode === ClientAuthErrorMessage.endpointResolutionError.code) {
```

Comment:
How could ` acquireTokenAuthority.tokenEndpoint` throw this error?

Reply:
if you review the code, the getter for the tokenEndpoint checks whether endpoint discovery is complete, and throws an error if it isn't. I can change this so that we check whether endpoint discovery is done before calling `acquireTokenAuthority.tokenEndpoint`.

Reply:
Yeah, missed that initially. Yeah, you would know when calling `createInstance`, correct? That would seem like a more intuitive place to throw.

Reply:
made changes